mobile/app/lib/cothority/network/proto.ts

```diff
@@ -172,21 +173,22 @@ export class ServerIdentity extends Message<ServerIdentity> {
      */
     static urlToWebsocket(url: string): string {
         const urlParser = new URL(url);
+        let proto = "";
         switch (urlParser.protocol) {
             case "http:": {
-                urlParser.protocol = "ws:";
+                proto = "ws:";
                 break;
             }
             case "https:": {
-                urlParser.protocol = "wss:";
+                proto = "wss:";
                 break;
             }
             default : {
                 throw new Error("The url field should use either 'http:' or 'https:', but we found "
                     + urlParser.protocol);
             }
         }
-        let result = urlParser.toString();
+        let result = url.replace(/^https?:/, proto);
```

Comment:
If you modify `urlParser` directly, no need to string replace stuff.

Reply:
That was the original `new URL`, but the `url-parser` doesn't allow to set the `protocol`, this is why I changed it here to use the `proto` variable.

Reply:
Hum, it seems that there is a [URL.set](https://www.npmjs.com/package/url-parse#urlsetkey-value) to do that, maybe something like that?
```suggestion
        urlParser.set("protocol", proto);
```